### PR TITLE
Add thread_name in system.stack_trace

### LINKED ISSUE #23256: Thread name in system.stack_trace

### PR DESCRIPTION
**Pull request by abel-cheng**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add `thread_name` column in `system.stack_trace`. This closes #23256.

Detailed description / Documentation draft:

add thread_name column in system.stack_trace
https://github.com/ClickHouse/ClickHouse/issues/23256

By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.

Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Storages/System/StorageSystemStackTrace.cpp [hunk 1]
```diff
@@ -213,6 +215,21 @@ void StorageSystemStackTrace::fillData(MutableColumns & res_columns, ContextPtr,
...
+        if (std::filesystem::exists(thread_name_path))
+        {
+            std::ifstream comm(thread_name_path.string());
+            std::string s;
+            while (thread_name && std::getline(comm, s))
+            {
+                thread_name << s;
+            }
+            comm.close();
+        }
```
Comment (tavplubix) [inline src/Storages/System/StorageSystemStackTrace.cpp:231]: Could you please replace `std::ifstream` and `std::ostringstream` with `ReadBufferFromFile` and `readStringUntilEOF(...)`?
(that's why Style Check failed)

Comment (abel-cheng) [reply src/Storages/System/StorageSystemStackTrace.cpp:231]: Thank you for your advice, I will change it soon.

**Review by tavplubix — APPROVED** [review]

**Comment by sevirov** [comment]
Internal documentation ticket: DOCSUP-9597.

**Comment by alexey-milovidov** [comment]
@tavplubix thread_name column has excessive newline at end. I will fix it.

**Pull request merged**